webdriverio/index.d.ts

```diff
@@ -30,96 +30,116 @@ declare namespace WebdriverIO {
 
     // Action
     export interface Client<T> {
-        addValue(selector: string, value: string | number): Client<void>;
+        addValue(
+            selector: string,
+            value: string | number
+        ): Client<void>;
+        addValue(value: string | number): Client<void>;
         addValue<P>(
             selector: string,
             value: string | number,
-            callback: (err: any) => P
         ): Client<P>;
+        addValue<P>(value: string | number): Client<P>;
 
         clearElement(selector: string): Client<void>;
-        clearElement<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        clearElement(): Client<void>;
+        clearElement<P>(selector: string): Client<P>;
+        clearElement<P>(): Client<P>;
 
         click(selector: string): Client<void>;
-        click<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        click(): Client<void>;
+        click<P>(selector: string): Client<P>;
+        click<P>(): Client<P>;
 
         doubleClick(selector: string): Client<void>;
-        doubleClick<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        doubleClick(): Client<void>;
+        doubleClick<P>(selector: string): Client<P>;
+        doubleClick<P>(): Client<P>;
 
-        dragAndDrop(sourceElem: string, destinationElem: string): Client<void>;
+        dragAndDrop(
+            sourceElem: string,
+            destinationElem: string
+        ): Client<void>;
+        dragAndDrop(destinationElem: string): Client<void>;
         dragAndDrop<P>(
             sourceElem: string,
-            destinationElem: string, callback: (err: any) => P
+            destinationElem: string
         ): Client<P>;
+        dragAndDrop<P>(destinationElem: string): Client<P>;
 
         leftClick(selector: string): Client<void>;
-        leftClick<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        leftClick(): Client<void>;
+        leftClick<P>(selector: string): Client<P>;
+        leftClick<P>(): Client<P>;
 
         middleClick(selector: string): Client<void>;
-        middleClick<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        middleClick(): Client<void>;
+        middleClick<P>(selector: string): Client<P>;
```

Comment:
You're absolutely right. I was so caught up in search/replace that I forgot to think about the sanity of doing as I did. selector? would be much cleaner. I'll fix it right away.